config/quality/checkstyle/checkstyle.xml

```diff
@@ -1,238 +1,240 @@
+        <module name="FinalParameters" />
+        <module name="Indentation" />
+        <module name="OuterTypeFilename" />
+        <module name="TodoComment">
+            <property name="format" value="(?i)\s+CR\s+" />
+            <property name="severity" value="info" />
+        </module>
+        <module name="TodoComment">
+            <property name="format" value="(?i)\s+TODO\s+" />
+            <property name="severity" value="info" />
+        </module>
+        <module name="TrailingComment" />
+        <module name="UncommentedMain" />
+        <module name="UpperEll" />
+
+        <!-- Modifier Checks                                    -->
+        <!-- See http://checkstyle.sourceforge.net/config_modifiers.html -->
+        <module name="ModifierOrder" />
+        <module name="RedundantModifier" />
+
         <!-- Checks for Naming Conventions.                  -->
         <!-- See http://checkstyle.sourceforge.net/config_naming.html -->
+        <module name="AbbreviationAsWordInName" />
+        <module name="AbstractClassName" />
+        <module name="CatchParameterName" />
+        <module name="ClassTypeParameterName" />
         <module name="ConstantName" />
+        <module name="InterfaceTypeParameterName" />
         <module name="LocalFinalVariableName" />
-        <module name="FinalLocalVariable">
-            <property name="tokens" value="VARIABLE_DEF,PARAMETER_DEF"/>
-        </module>
         <module name="LocalVariableName" />
         <module name="MemberName" />
-        <module name="MethodName">
-            <property name="severity" value="ignore" />
-        </module>
+        <module name="MethodName" />
+        <module name="MethodTypeParameterName" />
         <module name="PackageName" />
-        <module name="ParameterName">
-            <property name="severity" value="ignore" />
-        </module>
-        <module name="StaticVariableName">
-            <property name="severity" value="ignore" />
-        </module>
+        <module name="ParameterName" />
+        <module name="StaticVariableName" />
         <module name="TypeName" />
 
-        <!-- Checks for imports                              -->
-        <!-- See http://checkstyle.sourceforge.net/config_import.html -->
-        <module name="AvoidStarImport" />
-        <module name="IllegalImport" />
-        <!-- defaults to sun.* packages -->
-        <module name="RedundantImport" />
-        <module name="UnusedImports" />
-
+        <!-- Checks for Naming Conventions.                  -->
+        <!-- See http://checkstyle.sourceforge.net/config_naming.html -->
+        <module name="Regexp" />
 
         <!-- Checks for Size Violations.                    -->
         <!-- See http://checkstyle.sourceforge.net/config_sizes.html -->
+        <module name="AnonInnerLength">
+            <property name="max" value="20" />
+        </module>
+        <module name="ExecutableStatementCount" />
```

Comment:
Well, actually no. Checkstyle forces you to have your code in the same style as all other Java developers (who are using also Checkstyle hopefully). To have you code resembling the code of all other Java developers as much as possible you want to enable as much rules as possible except for the rules for which you have a very good reason to suppress them. With "when in doubt, leave it out" is meant something more difficult in this situation. The standard is that everything is included, that's the way to go, so if you doubt whether or not you should leave something out (ie you're in doubt if you should not ignore the rule), you should leave it out (ie follow the standard, have the last amount of code, don't add code to ignore the warning)